src/spikeinterface/preprocessing/silence_periods.py

```diff
@@ -71,28 +72,39 @@ def __init__(self, recording, list_periods, mode="zeros", noise_levels=None, **r
         if mode in ["noise"]:
             if noise_levels is None:
                 noise_levels = get_noise_levels(
-                    recording, return_scaled=False, concatenated=True, **random_chunk_kwargs
+                    recording, return_scaled=False, concatenated=True, seed=seed, **random_chunk_kwargs
                 )
+            noise_generator = NoiseGeneratorRecording(
+                num_channels=recording.get_num_channels(),
+                sampling_frequency=recording.sampling_frequency,
+                durations=[recording.select_segments(i).get_duration() for i in range(recording.get_num_segments())],
+                dtype=recording.dtype,
+                seed=seed,
+                noise_levels=noise_levels,
+                strategy="on_the_fly",
+                noise_block_size=int(recording.sampling_frequency),
+            )
```

Comment:
same here. Can you add a test in `test_silence`?

Reply:
A small test checking that noise is properly seeded has been added